packages/interactivity/src/directives.js

```diff
@@ -14,23 +14,118 @@ import { useWatch, useInit } from './utils';
 import { directive, getScope, getEvaluate } from './hooks';
 import { kebabToCamelCase } from './utils/kebab-to-camelcase';
 
-const isObject = ( item ) =>
-	item && typeof item === 'object' && ! Array.isArray( item );
+// Assigned objects should be ignore during proxification.
+const contextAssignedObjects = new WeakMap();
 
-const mergeDeepSignals = ( target, source, overwrite ) => {
+const isPlainObject = ( item ) =>
```

Comment:
This will similar as the one in `store.ts`
https://github.com/WordPress/gutenberg/pull/59039/

Should we move it to another file to share same code in different files?

Reply:
Yeah, maybe we could move it to `utils.js`. 🤔